New chmduquesne/rollinghash breaks tests

Hello!

Apparently we have a newer version of rollinghash in Debian (4.0.0) and it broke syncthing tests:
```
ok  	github.com/syncthing/syncthing/lib/relay/protocol	0.281s [no tests to run]
=== RUN   TestBlocks
--- PASS: TestBlocks (0.00s)
=== RUN   TestAdler32Variants
--- FAIL: TestAdler32Variants (0.01s)
	blocks_test.go:160: At i=128, sum2=5ea93e99, sum3=5ea93e99
	blocks_test.go:160: At i=256, sum2=2cf17ba3, sum3=80683d0b
	blocks_test.go:162: Mismatch after roll; i=256, sum2=2cf17ba3, sum3=80683d0b
```

This may be a misdiagnosis, I have only spent little time on this for now.

Let me know if you think the problem is related or not.

Cheers,

--- a/lib/scanner/blocks_test.go
+++ b/lib/scanner/blocks_test.go
@@ -144,14 +144,13 @@ func TestAdler32Variants(t *testing.T) {
 
 	windowSize := 128
 
-	hf2.Reset()
-
 	hf3 := rollingAdler32.New()
 	hf3.Write(data[:windowSize])
 
 	for i := windowSize; i < len(data); i++ {
 		if i%windowSize == 0 {
 			// let the reference function catch up
+			hf2.Reset()
 			hf2.Write(data[i-windowSize : i])
 
 			// verify that they are in sync with the rolling function

